rct229/rulesets/ashrae9012019/ruleset_functions/get_lab_zone_hvac_systems.py

```diff
@@ -0,0 +1,74 @@
+from pydash import map_, filter_
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_dict_of_zones_and_terminal_units_served_by_hvac_sys import (
+    get_dict_of_zones_and_terminal_units_served_by_hvac_sys,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_zone_target_baseline_system import (
+    SYSTEMORIGIN,
+    get_zone_target_baseline_system,
+)
+
+
+def get_lab_zone_hvac_systems(
+    rmd_b: str, rmd_p: str, climate_zone: str, is_leap_year: bool
+):
+    """
+    returns a list of HVAC systems serving only lab zones
+
+    Parameters
+    ----------
+    rmd_b: json
+        RMD at RuleSetModelDescription level
+    rmd_p: json
+        RMD at RuleSetModelDescription level
+    climate_zone: str
+        baseline climate zone
+    is_leap_year: bool
+        Whether leap year or not
+
+    Returns
+    -------
+    hvac_systems_serving_lab_zones: A dictionary consisting of two lists of hvac system ids.
+    One list ["lab_zones_only"] is a list of hvac system ids where each hvac system serves only lab zones.
+    The other ["lab_and_other"] is a list of hvac system ids where systems serve both labs and other zones.
+    """
+
+    target_baseline_systems = get_zone_target_baseline_system(
+        rmd_b, rmd_p, climate_zone, is_leap_year
+    )
+
+    # find zone ids whose `system_origin` is `SYSTEMORIGIN.G311D`
+    building_lab_zones = [
+        zone_id
+        for zone_id in target_baseline_systems
+        if target_baseline_systems[zone_id]["system_origin"] == SYSTEMORIGIN.G311D
+    ]
+
+    hvac_systems_serving_lab_zones = {"lab_zones_only": [], "lab_and_other": []}
+
+    if building_lab_zones:
+        dict_of_zones_and_hvac_systems = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmd_b)
+        )
+
+        for hvac_id_b in dict_of_zones_and_hvac_systems:
+            get_zone_ids = dict_of_zones_and_hvac_systems[hvac_id_b]["zone_list"]
+
+            # determine zone ids are included in building_lab_zones
+            zone_included_in_building_lab_zones = filter_(
+                get_zone_ids, lambda zone_id: zone_id in building_lab_zones
+            )
+            zone_included_in_building_lab_zones_bool = map_(
+                get_zone_ids, lambda zone_id: zone_id in building_lab_zones
+            )
+
+            # assign the zone ids
+            if all(zone_included_in_building_lab_zones_bool):
+                hvac_systems_serving_lab_zones[
+                    "lab_zones_only"
+                ] += zone_included_in_building_lab_zones
```

Comment:
This is incorrect. The RDS appends `hvac_id` to the list instead of the zone id lists.

Reply:
Good point. Addressed.